File: core/src/main/scala/eu/ostrzyciel/jelly/core/internal/RowBufferAppender.scala
package eu.ostrzyciel.jelly.core.internal

import eu.ostrzyciel.jelly.core.proto.v1.*

/**
 * Internal trait for appending lookup entries to the row buffer.
 *
 * This is used by NodeEncoder.
 */
private[core] trait RowBufferAppender:
  private[core] def appendNameEntry(entry: RdfNameEntry): Unit
  private[core] def appendPrefixEntry(entry: RdfPrefixEntry): Unit
  private[core] def appendDatatypeEntry(entry: RdfDatatypeEntry): Unit

File: core-java/src/main/java/eu/ostrzyciel/jelly/core/RowBufferAppender.java
package eu.ostrzyciel.jelly.core;

import eu.ostrzyciel.jelly.core.proto.v1.RdfDatatypeEntry;
import eu.ostrzyciel.jelly.core.proto.v1.RdfNameEntry;
import eu.ostrzyciel.jelly.core.proto.v1.RdfPrefixEntry;

/**
 * Interface for appending lookup entries to the row buffer.
 * <p>
 * This is used by NodeEncoder.
 */
public interface RowBufferAppender {
    void appendNameEntry(RdfNameEntry nameEntry);
    void appendPrefixEntry(RdfPrefixEntry prefixEntry);
    void appendDatatypeEntry(RdfDatatypeEntry datatypeEntry);
}
